bushes/init.lua

```diff
@@ -195,53 +153,108 @@ abstract_bushes.grow_bush_node = function(pos,dir, leaf_type)
 		dir = 1
 	end
 
-	if minetest.get_node(right_here).name == "air"	-- instead of check_air = true,
-	or minetest.get_node(right_here).name == "default:junglegrass" then
+	local nodename = minetest.get_node(right_here).name
+	if nodename == "air" or nodename == "default:junglegrass" then -- instead of check_air = true,
 		minetest.swap_node(right_here, {name="bushes:bushbranches"..bush_branch_type , param2=dir})
-						--minetest.chat_send_all("leaf_type: (" .. leaf_type .. ")")
 		minetest.swap_node(above_right_here, {name="bushes:BushLeaves"..leaf_type})
+
 		local chance_of_high_leaves = math.random(1,10)
-		if chance_of_high_leaves> 5 then
+		if chance_of_high_leaves > 5 then
 			local two_above_right_here = {x=pos.x, y=pos.y+3, z=pos.z}
-							--minetest.chat_send_all("leaf_type: (" .. leaf_type .. ")")
 			minetest.swap_node(two_above_right_here, {name="bushes:BushLeaves"..leaf_type})
 		end
 	end
 end
 
+local function grow_bush(pos)
+	-- replace possible grass nodes
+	minetest.swap_node({x=pos.x, y=pos.y+1, z=pos.z}, {name="air"})
```

Comment:
That seems maybe bad, but it fixes all "one node high floating bushes" and I haven't seen any broken trees.